garcon/activity.py

```diff
@@ -253,21 +255,37 @@ def run(self):
         previous activity is consumed (context).
         """
 
-        try:
-            activity_task = self.poll()
-        except Exception as error:
+        max_try = 5
+        for try_number in range(1, max_try+1):
```

Comment:
You need a `while` loop here. Consider the following scenario: 
- ThrottlingException has been thrown 4 times. 
- Things are working fine now for 2 days. 
- A single Throttle exception will apply an exponential backoff of 5.

So you want to be able to reset this number as soon as you are able to poll an activity task. A for loop wouldn't allow you to do this. 

```
while try_number < max_try:
    try:
       activity_task = self.poll()
       try_number = 0
    except Exception as error:
        if isTHrottlingException(error) and :
            sleep_time = utils.expenential_backoff_delay(2, try_number)
        ...
```